### PR TITLE
fix: Fixing navigation failures on volume boundaries edges

### PR DESCRIPTION
**Pull request by noemina**
Implements a way to hit boundaries when this is the only possible option. Closes https://github.com/acts-project/acts/issues/1720

### CODE REVIEW

**Comment by paulgessinger** [comment]
image

👎

**Comment by noemina** [comment]
> image
> 
> -1

Now fixed!

File: Core/src/Geometry/TrackingVolume.cpp [hunk 1]
```diff
@@ -484,6 +484,27 @@ Acts::TrackingVolume::compatibleBoundaries(
       return BoundaryIntersection();
     }
 
+    if (options.forceIntersectBoundaries and
+        sIntersection.intersection.pathLength * options.navDir > 0) {
```
Comment (andiwand) [inline Core/src/Geometry/TrackingVolume.cpp:488]: does this do the same as `std::abs(sIntersection.intersection.pathLength) >0`? in that case we could also check for `sIntersection.intersection.pathLength != 0` if you think that is preferable

**Review by andiwand — APPROVED** [review]

File: Core/src/Geometry/TrackingVolume.cpp (hunk 1 above)
Comment (noemina) [reply Core/src/Geometry/TrackingVolume.cpp:488]: This is to check that the boundary surface you want to intersect is along the direction, hence `sIntersection.intersection.pathLength` is required to have the same sign of `options.navDir`.

**Review by asalzburger — APPROVED** [review]

**Comment by asalzburger** [comment]
If this is actually needed, we might consider the default to be set to `true` ?

**Comment by noemina** [comment]
> If this is actually needed, we might consider the default to be set to `true` ?

@asalzburger , this can be set to `true` only if the previous checks fail.

**Pull request merged**